### DISCUSSION TITLE
Moving boundaries

### DISCUSSION THREAD

**Discussion by Apoorghani3** [ProjectPhysX/FluidX3D#156] [Q&A]
Hi
Impressive thing you got here!
I have a question about simulating moving boundaries on a pipe.
I want to solve Fluid-structure interaction FSI in a vessel.
Is this solver included in or Should I develop it?

**Comment by ProjectPhysX** [comment]
What exactly is your setup? Vessel in the sense of ship, or blood vessel?
I had a model for red blood cell membranes in an unreleased alpha version.
Moving boundaries currently are one-way interaction, the boundary movement is fixed and fluid acts accordingly. However it's possible to add to-way, by computing force/torque and moving boundaries accordingly.

**Reply by Apoorghani3** [reply]
That is exactly what I need, a two-way coupled fluid structure interaction.
It is a vessel with a deformable valve in it which deforms due to interaction with the fluid.
The vessel wall movement is prescribed (moving boundaries) but the deformation of solid valve inside the vessel must be calculated.

**Reply by ProjectPhysX** [reply]
If valve flaps are like solid doors with spring opening force, this should be doable: sum torque on valve flap around hinge, set opening angle proportional to torque, revoxelize, repeat. Force/torque is accurate in low-Reynolds regime.

**Reply by Apoorghani3** [reply]
here is a sample valve deformation

https://github.com/ProjectPhysX/FluidX3D/assets/162499995/351ca99c-a25d-4b3a-ab44-da4c26788cf3

it is not the flap like structure rotating around an axis, it is constrained to the vessel wall and it deforms due to fluid movement
